### PR TITLE
Polybar: fix quoting of config values, fixes #356

### LINKED ISSUE #356: Polybar module double-quotes values, causing unexpected behavior
I'm really new at Nix, so apologies in advance if this is off-base. 

When declaring key-value pairs for a Polybar config, using `polybar.config`, the module correctly generates a config file, but double-quotes the values. The offending line is here: 

https://github.com/rycee/home-manager/blob/2548c4317519f5b23c3d3304d69e12651da36e5a/modules/services/polybar.nix#L16

This would be fine, ordinarily, except that the INI file format used by Polybar doesn't quote values unless whitespace is really important to the value. 

This causes problems when you want to leave a field blank. For example, I want to set polybar's `label-empty` key to nothing, i.e. `""`. But the polybar module outputs literal double quotes, so that the workspaces that were previously hidden now all have the label `""`. 

The easy fix is just only quoting these values once. I'll submit a PR.

### PR DESCRIPTION
**Pull request by JonathanReeve**

### CODE REVIEW

**Comment by uvNikita** [comment]
Thank you for raising the issue and opening PR!

The reason for such quoting was exactly as the one you specified in the #356:

> Polybar doesn't quote values unless whitespace is really important to the value

E.g. I thought it would be counterintuitive if someone would have `option = "something   "` in their configuration and extra spaces in the end wouldn't be preserved.

In my configuration, for example, I was using `format-prefix` with spaces in the end for additional padding. Since then I, however, discovered [format-NAME-padding](https://github.com/jaagr/polybar/wiki/Formatting) option that solves the issue without using extra spaces hack.

I see two options here:
1. To add a special case for empty value. The benefit here is that it will be compatible with the current format and users will not have to change anything.
2. To go with your patch and suggest users to change their configuration to `option = ''"something   "''` in case they need it (I haven't tested, but I guess it should work).

I wouldn't mind going with the second option if we can be quite sure that padding/margin settings cover most cases when extra spaces are needed.

**Comment by rycee** [comment]
@JonathanReeve Thanks for the contribution, I have no strong opinion about this since I don't use this module myself. Although it would be nice, if possible, to keep backwards compatibility.

I mostly wanted to point out that the commit message to read something along the lines of

> polybar: single-quote instead of double
>
> Polybar expects quoted values only when whitespace is important to the value.
>
> Fixes #356

**Comment by JonathanReeve** [comment]
Is there a way to check (using some nix library) whether there's whitespace at either end of a value, and quote it if so? That way it follows the INI specification of only quoting when necessary, maintains backwards compatibility, and solves the empty value issue.

**Comment by rycee** [comment]
Something like
```patch
diff --git a/modules/services/polybar.nix b/modules/services/polybar.nix
index a187f62..423b0f1 100644
--- a/modules/services/polybar.nix
+++ b/modules/services/polybar.nix
@@ -11,9 +11,14 @@ let
   toPolybarIni = generators.toINI {
     mkKeyValue = key: value:
     let
+      quoted = v:
+        if hasPrefix " " v || hasSuffix " " v
+        then ''"${v}"''
+        else v;
+
       value' =
         if isBool value then (if value then "true" else "false")
-        else if (isString value && key != "include-file") then ''"${value}"''
+        else if (isString value && key != "include-file") then quoted value
         else toString value;
     in
       "${key}=${value'}";
```

might do it.

**Comment by uvNikita** [comment]
Nice! I think this should do the trick. If you can make the changes I'll be happy merge this PR :)

**Comment by JonathanReeve** [comment]
Awesome. I applied @rycee's patch. (And changed the commit message and PR message while I was at it.) Let me know if this looks good!

File: modules/services/polybar.nix [hunk 1]
```diff
@@ -10,10 +10,15 @@ let
 
   toPolybarIni = generators.toINI {
     mkKeyValue = key: value:
-    let
+   let
```
Comment (rycee) [inline modules/services/polybar.nix]: The indentation lost a space here 🙂

Comment (JonathanReeve) [reply modules/services/polybar.nix]: Oops! Just added a fix, I think.

**Comment by rycee** [comment]
Nice, thanks! I squashed it all and rebased to master in 629d66e0b983b0ea5780ddf805f65a024d7deb5b.

**Comment by uvNikita** [comment]
@JonathanReeve thanks for the contribution! And thank you for merging it @rycee :)

**Pull request closed without merging**